src/components/RangesDisplay.js

```diff
@@ -28,8 +19,15 @@ export const RangesDisplay = () => {
     const rangeRadius = radius + filledArcWidth;
     const centerX = rangeRadius + margin;
     const centerY = rangeRadius + margin;
-    const end = arcsStart + index * (arcSegmentDegree + spaceBetweenSegments);
-    const start = arcSegmentDegree + end;
+
+    const arc = arcs[index];
+    const previousArc = arcs[index - 1];
+
+    let end = 0;
+    if (previousArc) end = previousArc.end + spaceBetweenSegments;
+
+    let start = end;
+    if (arc) start = arc.end;
```

Comment:
This was very confusing until I realized the start and end are kind of reversed. I believe it's due to the `textAnchor` enum on the RNSVG Text component.